File: cours-02/egalites.py
# Égalités en Python.

class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y
        
    def __eq__(self, obj):
        if isinstance(obj, Point):
            return self.x == obj.x and self.y == obj.y
        return False
        
    def move(self, dx, dy):
        self.x += dx
        self.y += dy
        
    def print(self):
        print(f"({self.x}, {self.y})")
        
p = Point(2, 4)
p2 = Point(2, 4)

print(p == p2)

File: cours-02/egalites.java
// Égalités en Java

public class Main
{
    public static class Point {
        private int x;
        private int y;
        
        public Point() { 
            x = 0;
            y = 0;
        }
        
        public Point(int x, int y) {
            this.x = x;
            this.y = y;
        }
        
        @Override
        public boolean equals(Object obj) {
            if(obj instanceof Point) {
               Point p = (Point)obj;
               return x == p.x && y == p.y;
            } else {
                return false;
            }
        }
        
        @Override
        public int hashCode() {
            // ...
        }
        
        public void move(int dx, int dy) {
            x += dx;
            y += dy;
        }
        
        public void print() {
            System.out.println("(" + x + ", " + y + ")");
        }
    }
    
	public static void main(String[] args) {
	    Point p = new Point(3, 4);
	    Point p2 = new Point(3, 4);
	    
	    p.print();
	    p2.print();
	    
	    System.out.println(p.equals(p2));
	}
}